core/trino-main/src/main/java/io/trino/sql/planner/plan/Assignments.java

```diff
@@ -141,19 +141,12 @@ public boolean isIdentity(Symbol output)
     {
         Expression expression = assignments.get(output);
 
-        return expression instanceof SymbolReference && ((SymbolReference) expression).getName().equals(output.getName());
+        return expression instanceof SymbolReference symbolReference && symbolReference.getName().equals(output.getName());
     }
 
     public boolean isIdentity()
     {
-        for (Map.Entry<Symbol, Expression> entry : assignments.entrySet()) {
-            Expression expression = entry.getValue();
-            Symbol symbol = entry.getKey();
-            if (!(expression instanceof SymbolReference && ((SymbolReference) expression).getName().equals(symbol.getName()))) {
-                return false;
-            }
-        }
-        return true;
+        return assignments.keySet().stream().allMatch(this::isIdentity);
```

Comment:
This way incurs map look-ups while the current code is just iterating over all values

Reply:
The increase look-ups only when the `value` is `identity` and will call the `map.get`, I think the cost may can be ignore, but improve the readability, WDYT?

Reply:
planner is performance sensitive and the older code didn't have bad readability anyway.

Reply:
Alright, I was confused at the first glance, thanks